src/msg_recorder/recorder.cc

```diff
@@ -0,0 +1,56 @@
+#include "cris/core/msg_recorder/recorder.h"
+
+#include "cris/core/utils/logging.h"
+
+#include "impl/utils.h"
+
+#include <filesystem>
+#include <sstream>
+
+namespace cris::core {
+
+MessageRecorder::MessageRecorder(const std::filesystem::path& record_dir_prefix, std::shared_ptr<JobRunner> runner)
+    : Base(std::move(runner))
+    , record_dir_(record_dir_prefix / RecordDirNameGenerator()) {
+    std::filesystem::create_directories(record_dir_);
+}
+
+MessageRecorder::~MessageRecorder() {
+    files_.clear();
+
+    if (std::filesystem::is_empty(GetRecordDir())) {
```

Comment:
It'll leak if killed. You need a gc mechanism for this.

Reply:
This is not for GC but more like an optimization. It removes the empty records. If it is killed then there will be an empty record left and I think that is OK and we don't need to do more here. We may have some postprocessing checks when collecting the records.

Reply:
> We may have some postprocessing checks when collecting the records.

No, you just need a boot time fsck

Reply:
The dir will be different in the next boot time.